### ISSUE TITLE
global .hgignore file is ignored

### ISSUE THREAD

**Issue by hyOzd** [ggreer/the_silver_searcher#954]
Ag doesn't respect global `~/.hgignore` file.

I want to mention that this is not the standard file name though, it is usually set in users configuration (.hgrc) like this:

    [ui]
    ignore = ~/.hgignore

And can be queried with this command:

     hg config ui.ignore

It would be nice to have this supported.

**Comment by bmalehorn** [comment]
`ag` doesn't rely on any existing binaries on the system, such as `hg` or `git`. Using something like `hg config ui.ignore` would require that to be installed.

Assuming `hg` is installed, we could run `hg locate` in mercurial directories to get 100% accurate ignores. We wouldn't need special logic for parsing `.hgignore` files or applying their ignores.

But, there are a handful of downsides to relying on `hg`:
- If `hg` is not installed, how do we get our ignores? Do we forget about them and search every file?
- If `hg` is installed but the repo is in some broken / strange state, `ag` will crash
- If `hg locate` is slow, `ag` will be slow
- Is `hg` now a dependency of `ag` in apt / brew / macports / yum / packman?

tl;dr - `hg config ui.ignore` is not going to happen

But an OK solution would be to blindly assume the global ignore would be located at `~/.hgignore`. I bet this guess will be correct for 95% of global ignores, which is a lot better than the 0% we're getting right now.

**Comment by hyOzd** [comment]
Hmm, I agree with your concerns. Assuming the file is named `~/.hgignore` doesn't sound like a bad idea now.
